### ISSUE TITLE
Details about fullMutationMatrix

### ISSUE THREAD

**Issue by oscarlvp** [STAMP-project/descartes-issue-generator#1]
Unless you need the **entire** list of failing tests, you don't need to use `fullMutationMatrix`. The `XML` and `JSON` reports provide the list of ordered tests that PIT will execute and how many of them were executed. So in order to find one failing test you don't need to compute the full matrix. It is the first reported test after all the executed tests.
Computing the matrix is highly time consuming.
`JSON` report supports the full matrix, since Descartes 1.2.5 see the [release notes](https://github.com/STAMP-project/pitest-descartes/releases/tag/descartes-1.2.5)

**Comment by gibello** [comment]
In fact, what I need is to find succeeding tests, not failing ones (I opened an issue in PIT because MutationResult.getSucceedingTests() was always empty... Henry Coles answered that "Succeeding tests are only recorded when pitest is run in "test matrix" mode" ).

Maybe I could find at least one succeeding test without fullMutationMatrix ? If so, full matrix could become optional (more accurate and complete, but slower).

Any idea how I could find one or more succeeding test(s) without full matrix ? Would like to implement that...

**Comment by oscarlvp** [comment]
When you run without `fullMutationMatrix`, if the mutant has status `SURVIVED`, all test cases in `MutationResult. getTestsInOrder` pass. That is actually what this status means: all test cases pass. So there is no need to run the full matrix.
Running the full matrix actually makes no difference in those mutants. The full matrix does something different when the mutant is `KILLED`. The usual PIT behavior is to stop when the first test case fails for a mutant. With `fullMutationMatrix` all tests are executed no matter if they fail or not.
To summarize, what you need, when the mutant `SURVIVED` are the test cases returned by `getTestsInOrder`.

**Comment by gibello** [comment]
Seems to work.
In fact, with "full matrix", we know when a mutation is detected by only some of the tests, not all of them. So the report is more accurate.
Without "full matrix", we only get undetected mutations (= detected by no test at all).
I will have to document that (it makes "full matrix" a valuable option, unless you need more performance).

**Comment by oscarlvp** [comment]
It is true that the full matrix provides more information, but in the end we evaluate the complete test suite as a set of test cases. If the mutant is killed, then the test suite as a whole is able to detect it. Not all test cases should kill all mutants. Probably developers, and for sure DSpot, only need to know if the mutant survived or not to the entire test suite. But as you say, it is really good to have the choice on whether the full matrix should be computed if they want more details.
Since it is really expensive, I don't think it should be enabled by default.
